frontend/src/metabase-lib/metadata.ts

```diff
@@ -202,3 +190,16 @@ export function fromLegacyColumn(
 ): ColumnMetadata {
   return ML.legacy_column__GT_metadata(query, stageIndex, columnOrField);
 }
+
+/**
+ * Even though it seems weird to pass the same query two times,
+ * this function follows the same pattern as the other displayInfo functions.
+ * The first two parameters are always a query, and a stage.
+ * The third parameter is what you would like to have info about.
+ * It just only happens that the thing we're examining is (again) the query itself.
+ *
+ * Rather than adding another overload, we're introducing a special-case, named abstraction.
+ */
+export function queryDisplayInfo(query: Query): QueryDisplayInfo {
+  return ML.display_info(query, -1, query);
```

Comment:
This comment applies to `ML.display_info` call, not to our `queryDisplayInfo`, so I'd move it inside.

Also:
`this function follows the same pattern as the other displayInfo functions` -> `this overload follows the same pattern as the other display_info overloads.`